src/NewsService.Validation/EditNewsRequestValidator.cs

```diff
@@ -0,0 +1,194 @@
+        {
+          { x => !string.IsNullOrEmpty(x.value.ToString()), "Preview cannot be empty." },
+        });
+
+      AddFailureForPropertyIf(
+        nameof(EditNewsRequest.Content),
+        x => x == OperationType.Replace,
+        new()
+        {
+          { x => !string.IsNullOrEmpty(x.value.ToString()), "Content cannot be empty." },
+        });
+
+      AddFailureForPropertyIf(
+        nameof(EditNewsRequest.Subject),
+        x => x == OperationType.Replace,
+        new()
+        {
+          { x => !string.IsNullOrEmpty(x.value.ToString()), "Subject cannot be empty." },
+          { x => x.value.ToString().Length < 120, "Subject is too long." },
+        });
+
+      AddFailureForPropertyIf(
+        nameof(EditNewsRequest.Pseudonym),
+        x => x == OperationType.Replace,
+        new()
+        {
+          { x => x.value.ToString().Length < 30, "Pseudonym is too long." },
+        });
+
+      #endregion
+
+      #region AuthorId, DepartmentId
+
+      AddFailureForPropertyIf(
+          nameof(EditNewsRequest.DepartmentId),
+          x => x == OperationType.Replace,
+          new()
+          {
+            { x => Guid.TryParse(x.value.ToString(), out Guid result), "Department id has incorrect format" },
+            { x => CheckDepartmentExistence(new List<Guid> { Guid.Parse(x.value.ToString()) }), "This department doesn't exist." }
+          });
+
+      AddFailureForPropertyIf(
+        nameof(EditNewsRequest.AuthorId),
+        x => x == OperationType.Replace,
+        new()
+        {
+          { x => Guid.TryParse(x.value.ToString(), out Guid result), "User id has incorrect format" },
+          { x => CheckUserExistence(new List<Guid> { Guid.Parse(x.value.ToString()) }), "This user doesn't exist." }
+        });
+
+      #endregion
+
+      #region IsActive
+
+      AddFailureForPropertyIf(
+        nameof(EditNewsRequest.IsActive),
+        x => x == OperationType.Replace,
+        new()
+        {
+          { x => bool.TryParse(x.value?.ToString(), out _), "Incorrect is active format." },
+        });
+
+      #endregion
+    }
+    public EditNewsRequestValidator(
+      IRequestClient<ICheckUsersExistence> rcCheckUsersExistence,
+      IRequestClient<ICheckDepartmentsExistence> rcCheckDepartmentsExistence,
+      ILogger<CreateNewsRequestValidator> logger)
+    {
+      _rcCheckUsersExistence = rcCheckUsersExistence;
+      _rcCheckDepartmentsExistence = rcCheckDepartmentsExistence;
+      _logger = logger;
+
+      RuleForEach(x => x.Operations)
+        .Custom(HandleInternalPropertyValidation);
+    }
+
+    private bool CheckUserExistence(List<Guid> authorsIds)
+    {
+      if (!authorsIds.Any() || authorsIds == default)
+      {
+        return false;
+      }
+
+      try
+      {
+        Response<IOperationResult<ICheckUsersExistence>> response =
+          _rcCheckUsersExistence.GetResponse<IOperationResult<ICheckUsersExistence>>(
+          ICheckUsersExistence.CreateObj(authorsIds)).Result;
```

Comment:
can you make it async and use await instead Result? If we cannot do it because constructor cannot be async so I think we have to move this check to command

Reply:
@MariyaSin  said to leave it as it is for now